plugins/uast-kotlin-fir/src/org/jetbrains/uast/kotlin/declarations/FirKotlinUMethod.kt

```diff
@@ -141,11 +143,54 @@ class FirKotlinConstructorUMethod(
     val isPrimary: Boolean
         get() = sourcePsi is KtPrimaryConstructor || sourcePsi is KtClassOrObject
 
+    override val uastBody: UExpression? by lz {
+        val delegationCall: KtCallElement? = sourcePsi.let {
+            when {
+                isPrimary -> ktClass?.superTypeListEntries?.firstIsInstanceOrNull<KtSuperTypeCallEntry>()
+                it is KtSecondaryConstructor -> it.getDelegationCall()
+                else -> null
+            }
+        }
+        val bodyExpressions = getBodyExpressions()
+        if (delegationCall == null && bodyExpressions.isEmpty()) return@lz null
+        KotlinLazyUBlockExpression(this) { uastParent ->
+            SmartList<UExpression>().apply {
+                // TODO: function call for delegationCall
+                bodyExpressions.forEach {
+                    add(baseResolveProviderService.baseKotlinConverter.convertOrEmpty(it, uastParent))
+                }
+            }
+        }
+    }
```

Comment:
Here there are no fir-spefic things too. Why isn't it in a common UAST part?

Reply:
Same here; there is nothing specific to FIR, except for function call expression for delegation call. It's unclear how (non-)trivial it would be to commonize function call, so I'd leave it in an abstract builder.

Unlike `UClass` commonization, `UMethod` has a tricky one: parameters. They will need different logic to resolve, which can be part of the base service, but I'm still not sure about _fake_ PSIs and desugaring (e.g., elvis, catch, `for`-loop). Until then, they will remain as different forms in different plugins. Except for parameters, though, we can commonize `UMethod`.

Another tricky part is that constructors are subtypes of abstract/base `UMethod`, and we want to share constructor-specific logic across plugins, while `UMethod` abstraction also needs plugin-specific implementation for aforementioned parameter stuff. By making such parameter creation as plugin-specific interface, even `UMethod`-to-constructors hierarchy can be commonized. The downside is, every concrete declaration needs to override parameter creations. :\